### ISSUE TITLE
Accelerated mode for Python3.5 or higher fails to import some extension modules linked against "Python3.dll"

### ISSUE THREAD

**Issue by kayhayen** [kayhayen/Nuitka#84]
When linking on Windows for Python3.5 or higher, there are now two DLLs and LIBs to choose from:

Those will be e.g. Python3.DLL and Python36.DLL and they both appear usable. For the accelerated binary it would be best to depend on a version specific DLL, ideally from the Python installation, but that has not yet been achieved. For standalone it is solved.

Then what can happen is that a "Qt.pyd" depends on Python3.DLL which it cannot find, and fails to load, although it normally always would.

**Comment by kayhayen** [comment]
So trying to get the DLL directory loaded, failed. I cannot master manifest files good enough to make it search for the Python36.DLL and (and then Python3.DLL) in the folder. Absolute paths might not be allowed anyway, so that is out of the question.

For the "Python3.DLL" however, we could make a determination where the Python36.DLL came from, and then add that as a DLL directory too. We already do that for the binary directory so it works in standalone mode, and for accelerated mode, it could be the directory where the PythonXX.DLL came from, which ought to be harmless. For a while already, we push something to "build_definitions.h" which is unused:

`#define DLL_EXTRA_PATH "c:\\python36_64"`

Could be time to start using it.  Adding it, things work for PyQt5.Qt, as it comes later. I still need to check if standalone mode picks it up too, as it is a dynamic solution in "main.exe" that it won't see.

**Comment by brupelo** [comment]
You didn't confirm me on the other thread but these issue you're having about python3.dll and python36.dll suggests me you were still using an old broken virtualenv version, am I right?

**Comment by kayhayen** [comment]
No, actually it is normal to have both Python3.DLL and Python3X.DLL. The first is redirecting to the later, but only a part. I think this is part of the effort to have a more portable API which is safe to use across versions. It's only a subset I believe too. I never bothered with it, but to make a wheel that will work on both 3.5 and 3.6, that would be the way to go.

Nothing broken about that. Indeed this is normal situation. It's only that I haven't come across this situation and they would also work for standalone previously, but not for accelerated mode, where the Python DLLs are not visible in the system.

**Comment by brupelo** [comment]
Indeed, it's correct to have both python3.dll and python3x.dll living in your python location (either system-wide or virtualenv). Thing is, there was a bug with older versions of virtualenv that wasn't copying both dlls (python3.dll and python3x.dll) to the right folders and that would make stuff like pyqt failing, take a look to this 1-min short [video](https://dl.dropboxusercontent.com/s/5fsq4zxx30x23zy/2018-05-11_01-21-53.mp4) to see what I mean by that.

**Comment by brupelo** [comment]
> Ah I see, so they caused issues to other things too with that.

Exactly, I had patched virtualenv to handle this situation as few months ago I've got the same problems than this [user](https://www.riverbankcomputing.com/pipermail/pyqt/2017-March/039032.html) using pyqt with the older virtualenv broken version.
